packages/destructure/src/index.ts

```diff
@@ -75,18 +81,24 @@ export function destructure<T extends ReactiveSource, O extends DestructureOptio
   source: MaybeAccessor<T>,
   options?: O,
 ): O extends { lazy: true; deep: true }
-  ? DeepDestructure<T>
-  : O["lazy"] extends true
-  ? Destructure<T>
+  ? DeepDestructure<T, O["memo"] extends "normalize" ? true : false>
+  : O extends { lazy: true }
+  ? Destructure<T, O["memo"] extends "normalize" ? true : false>
   : O["deep"] extends true
-  ? DeepSpread<T>
-  : Spread<T> {
+  ? DeepSpread<T, O["memo"] extends "normalize" ? true : false>
+  : Spread<T, O["memo"] extends "normalize" ? true : false> {
   const config: DestructureOptions<T> = options ?? {};
   const memo = config.memo ?? typeof source === "function";
-  const getter =
-    typeof source === "function"
-      ? (key: any) => () => source()[key]
-      : (key: any) => () => source[key];
+  const getter = (key: any) => {
+    const value = () => (typeof source === "function" ? source()[key] : source[key]);
+    if (config.memo !== "normalize") return value;
```

Comment:
no need to check the config and if the source is a function every time a property is accessed

Reply:
moved that into a createMemo outside of the getter.